apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/TimestampBound.cs

```diff
@@ -28,26 +28,26 @@ namespace Google.Cloud.Spanner.Data
     /// </summary>
     public sealed class TimestampBound : IEquatable<TimestampBound>
     {
-        private TimestampBound(TimestampBoundMode mode)
-        {
-            Mode = mode;
-        }
+        // Determines whether a read-only transaction will return the read timestamp.
+        private bool _returnReadTimestamp = false;
 
-        private TimestampBound(TimestampBoundMode mode, TimeSpan staleness)
+        private TimestampBound(TimestampBoundMode mode, TimeSpan staleness,
+                               DateTime timestamp, bool returnReadTimestamp = false)
         {
-            if (staleness.Ticks < 0)
+            if (staleness != TimeSpan.Zero && staleness.Ticks < 0)
```

Comment:
No need for the extra check here - TimeSpan.Zero.Ticks is 0 anyway, so it won't throw.